File: python/weighted_lottery_function.py
import random

weights_1 = {
    'win': 1,
    'break even': 5,
    'lose': 6
}


def weighted_lottery(weights):
    hat_container = []

    for outcome, odds in weights.items():
        for _ in range(odds):
            hat_container.append(outcome)
    random_outcome = random.choice(hat_container)
    print(f'You {random_outcome}!')

    play_again = (input('Play again? Yes/No:\n'))

    if play_again.lower().strip() == 'yes':
        weighted_lottery(weights)
    else:
        print('K, thx, bye.')


weighted_lottery(weights_1)

File: JavaScript/weightedLotteryFunction.js
const someWeights = {
    win: 1,
    draw: 4,
    lose: 5
};

const weightedLottery = function(weights) {
    const weightsArray = [];
    for (const weight of Object.entries(weights)) {
        for (let i = 0; i < weight[1]; i++) {
          weightsArray.push(weight[0]);
        }
    };
    const solution = weightsArray[Math.floor(Math.random() * weightsArray.length)];
    return solution;
};

console.log(weightedLottery(someWeights));